x-pack/plugins/security_solution/public/rules/routes.tsx

```diff
@@ -49,6 +55,21 @@ const RulesContainerComponent: React.FC = () => {
     <PluginTemplateWrapper>
       <TrackApplicationView viewId={SecurityPageName.rules}>
         <Switch>
+          {/* Because '/rules/id/:detailName/edit' would match '/rules/id/:detailName/:tabName' need
+          to add a redirect to ensure that it continues to work as expected */}
+          <Route
+            path="/rules/id/:detailName"
+            exact
+            strict
+            render={({ location: { search = '' }, match: { params } }) => (
+              <Redirect
+                to={{
+                  pathname: `/rules/id/${params.detailName}/alerts`,
+                  search,
+                }}
+              />
```

Comment:
Should we pass to the redirect also current query arguments, like `sourcerer`, etc.?

Reply:
So `search` corresponds to all that. It comes out to something like `'?sourcerer=(default:(id:security-solution-default,selectedPatterns:!(.alerts-security.alerts-default)))&timerange=(global:(linkTo:!(timeline),timerange:(from:%272022-08-29T07:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272022-08-30T06:59:59.999Z%27,toStr:now%2Fd)),timeline:(linkTo:!(global),timerange:(from:%272022-08-29T07:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272022-08-30T06:59:59.999Z%27,toStr:now%2Fd)))'`.

What I'm unsure of is why on the redirect, `search` is coming back as an empty string.

Reply:
`sourcerer` is a global URL param. It is saved in the redux store and preserved across all pages. The URL param gets deleted because Management pages opt out by setting the config `skipUrlState: true`.

You can read more about it here:  https://elastic.slack.com/archives/C012TJZ72J1/p1661845599185399?thread_ts=1661807452.300439&cid=C012TJZ72J1

Reply:
Just updating after our convo with Sergi on slack - we were using a deprecated method of adding anchor links to link to the rule details page. Updated the code to use the latest method.